src/test/java/com/synopsys/integration/jenkins/detect/substeps/ParseDetectArgumentsTest.java

```diff
@@ -0,0 +1,188 @@
+
+        assertEquals(8, detectCommandLine.size()); // Jar args (2) + passed args (2) + auto added args (3)
+        assertEquals(expectedJavaHome, detectCommandLine.get(0));
+        assertEquals("-jar", detectCommandLine.get(1));
+        assertEquals(expectedDetectPath, detectCommandLine.get(2));
+        commonValidation(detectCommandLine, "default");
+    }
+
+    @Test
+    public void testParseDetectArgumentsCustomLogLevel() {
+        expectedVisibleDetectProperties.put(loggingLevelKey, expectedCustomLogLevel);
+
+        DetectSetupResponse detectSetupResponse = new DetectSetupResponse(DetectSetupResponse.ExecutionStrategy.JAR, expectedJavaHome, expectedDetectPath);
+        ParseDetectArguments parseDetectArguments = new ParseDetectArguments(jenkinsIntLogger, intEnvironmentVariables, jenkinsVersionHelper, detectSetupResponse, createDetectPropertiesInputString());
+        List<String> detectCommandLine = parseDetectArguments.parseDetectArguments();
+
+        assertEquals(8, detectCommandLine.size()); // Jar args (2) + passed args (2) + auto added args (3)
+        assertEquals(expectedJavaHome, detectCommandLine.get(0));
+        assertEquals("-jar", detectCommandLine.get(1));
+        assertEquals(expectedDetectPath, detectCommandLine.get(2));
+        commonValidation(detectCommandLine, expectedCustomLogLevel);
+    }
+
+    @Test
+    public void testParseDetectArgumentsUnknownJenkinsVersion() {
+        Mockito.when(jenkinsVersionHelper.getJenkinsVersion()).thenReturn(java.util.Optional.empty());
+        Mockito.when(jenkinsVersionHelper.getPluginVersion("blackduck-detect")).thenReturn(java.util.Optional.empty());
+
+        DetectSetupResponse detectSetupResponse = new DetectSetupResponse(DetectSetupResponse.ExecutionStrategy.JAR, expectedJavaHome, expectedDetectPath);
+        ParseDetectArguments parseDetectArguments = new ParseDetectArguments(jenkinsIntLogger, intEnvironmentVariables, jenkinsVersionHelper, detectSetupResponse, "");
+        List<String> detectCommandLine = parseDetectArguments.parseDetectArguments();
+
+        assertEquals(6, detectCommandLine.size()); // Jar args (3) + passed args (0) + auto added args (3)
+        assertTrue(detectCommandLine.stream().anyMatch(argument -> argument.contains(jenkinsVersionParam + "=<unknown>")), errorMessage + "correct jenkins version");
+        assertTrue(detectCommandLine.stream().anyMatch(argument -> argument.contains(pluginVersionParam + "=<unknown>")), errorMessage + "correct plugins version");
+    }
+
+    private void commonValidation(List<String> detectCommandLine, String logLevelSearchValue) {
+        // This is added automatically when parseDetectArguments() is called
+        // The key will either be retrieved from the logger, or passed in as a parameter
+        expectedVisibleDetectProperties.put(loggingLevelKey, (logLevelSearchValue.equals("default") ? jenkinsIntLogger.getLogLevel().toString() : logLevelSearchValue));
```

Comment:
Since this is always going to be debug except for in testParseDetectArgumentsCustomLogLevel, perhaps the exception should modify the expectedVisibleDetectProperties instead of having every test do it.